design/ceph/ceph-config-via-cluster-crd.md

```diff
@@ -0,0 +1,53 @@
+# Ceph Config via Ceph Cluster CRD
+
+## Goal
+
+Adding the ability to configure the Ceph config options via the CephCluster CRD.
+The goal is not to replace the `rook-config-override` ConfigMap since it is still needed in some scenarios such as if mons need settings applied at first launch.
+
+## Current Solution
+
+Users need to update the `rook-config-override` and add their custom options, e.g., for setting a custom OSD scrubbing schedule ([Ceph OSD Scrubbing config options](https://docs.ceph.com/en/latest/rados/configuration/osd-config-ref/#scrubbing)).
+
+## Proposed Solution
+
+Adding a new structure to the CephCluster CRD under `.spec` named `cephConfig:`.
+
+The `cephConfig` structure will be structured in a way of `Target -> Options` (target being the service to set the options for, e.g., whole cluster `global`, specific OSD `osd.3`).
```

Comment:
I also had a late-night thought last night that we might want to keep an internal list of configs that Rook doesn't allow to be modified. This would be configs that are tightly controlled by Rook and that are critical to keeping the cluster functioning.

E.g., metrics values, mon configs, fsid, keyring, and any others I'm sure I missed.
@travisn @galexrt ?

Reply:
This feature is quite advanced and powerful, really making it easier to shoot yourself in the foot. My thought is that they need to be careful with this feature because we won't be able to put enough safeguards around crazy things they can do. I would vote not to put these safeguards in place and have it be a "use at your own risk" feature.

Reply:
@BlaineEXE That's a good point, I will amend the design in the implementation PR if that is fine.

Reply:
@travisn @BlaineEXE I hav thought this a bit more, I think it is reasonable to block the few most important settings (fsid, keyring, mon hosts), but to keep it at that.
In the future we might either remove that list or add a "I am really sure what I'm doing with these config options" option to disable this protection. Any thoughts on this?

Reply:
I would still vote to keep the implementation simple by not trying to restrict settings. Advanced features like this should be only used by advanced users who are careful about not choosing the wrong settings.

Reply:
I don't really see this as necessarily being an advanced-only feature, which is my struggle. I think that a lot of users will use this instead of using the config-override configmap. With that in mind, I do think that it is worth identifying a short list of truly critical configs that Rook should disallow users from modifying.

And I was originally thinking that the list should be the smallest and most critically small configs like @galexrt has suggested so that we can all keep our own sanity. Certainly `mon_hosts`, `fsid`, and `keyring` are likely the most core set.

Reply:
I'm not against putting those checks if you think it will help. There are so many settings that could be used incorrectly, it just feels like building a picket fence to keep out an elephant. :)